cmd/crank/render/runtime_docker.go

```diff
@@ -145,39 +154,85 @@ func GetRuntimeDocker(fn pkgv1.Function, log logging.Logger) (*RuntimeDocker, er
 	}
 	r := &RuntimeDocker{
 		Image:      fn.Spec.Package,
+		Name:       "",
 		Cleanup:    cleanup,
 		PullPolicy: pullPolicy,
 		log:        log,
 	}
 	if i := fn.GetAnnotations()[AnnotationKeyRuntimeDockerImage]; i != "" {
 		r.Image = i
 	}
+	if i := fn.GetAnnotations()[AnnotationKeyRuntimeNamedContainer]; i != "" {
+		r.Name = i
+	}
 	return r, nil
 }
 
 var _ Runtime = &RuntimeDocker{}
 
-// Start a Function as a Docker container.
-func (r *RuntimeDocker) Start(ctx context.Context) (RuntimeContext, error) {
-	r.log.Debug("Starting Docker container runtime", "image", r.Image)
-	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+func (r *RuntimeDocker) findContainer(ctx context.Context, cli *client.Client) (string, string) {
+	r.log.Debug("searching for Docker container", "name", r.Name)
+
+	filterArgs := filters.NewArgs()
+	filterArgs.Add("name", r.Name)
+
+	containers, err := cli.ContainerList(ctx, container.ListOptions{
+		Filters: filterArgs,
+		All:     true, // Include stopped containers
+	})
 	if err != nil {
-		return RuntimeContext{}, errors.Wrap(err, "cannot create Docker client using environment variables")
+		return "", ""
+	}
+
+	if len(containers) == 0 || containers[0].ID == "" {
+		r.log.Debug("no valid named container found", "name", r.Name)
+		return "", ""
+	}
+
+	for _, Container := range containers {
```

Comment:
Nit: `Container` is an unusual variable name for Go. I'd recommend `c`.

https://github.com/crossplane/crossplane/tree/main/contributing#use-descriptive-variable-names-sparingly